{
  final boolean forceToThrowException=node.groupProperties.FORCE_THROW_INTERRUPTED_EXCEPTION.getBoolean();
  final String error=Thread.currentThread().getName() + " is interrupted! Operation: " + op;
  if (safeToThrowException || forceToThrowException) {
    throw new RuntimeInterruptedException(error);
  }
 else {
    logger.log(Level.WARNING,error + " (To throw exception on interruption set '" + GroupProperties.PROP_FORCE_THROW_INTERRUPTED_EXCEPTION+ "' to true.)");
  }
}
